release/cc/create_artifacts.py

```diff
@@ -0,0 +1,184 @@
+# Copyright 2016 The Bazel Authors. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Module to create cc toolchain artifacts including targets and METADATA."""
+
+import imp
+import os
+import shlex
+import shutil
+from string import Template
+import subprocess
+
+from util import get_autoconfig_target_name
+from util import get_date
+from util import get_git_root
+
+BUILD_EXTRA_LICENCE = """
+licenses(["notice"])  # Apache 2.0
+
+package(default_visibility = ["//visibility:public"])
+"""
+
+GIT_ROOT = get_git_root()
+LICENCE_TPL = os.path.join(GIT_ROOT, "release", "license.tpl")
+CPP_TPL = os.path.join(GIT_ROOT, "release", "cc", "cpp.tpl")
+SHA_MAP_FILE = os.path.join(GIT_ROOT, "rules/toolchain_containers.bzl")
+
+
+def create_targets(container_configs_list, bazel_version):
+  """Create the new docker_toolchain_autoconfig target if not exists.
+
+  An example target located in configs/ubuntu16_04_clang/BUILD is:
+  //configs/ubuntu16_04_clang:msan-ubuntu16_04-clang-1.0-bazel_0.15.0-autoconfig
+
+  There is one target per container per Bazel version per config type.
+
+  The script only creates new targets in the BUILD file if they do not exist.
```

Comment:
Does this mean that if a target for the given version of bazel already exists it does not re-create it? (specify if so)

Reply:
Done.